### PR TITLE
feat: mcli integration

### PR DESCRIPTION
**Pull request by gssbzn**
## Description

Add support for `MCLI_*` env variables for convenience when using both tools 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have read the Terraform contribution guidelines
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

## Further comments

Also includes a couple of small bug fixes when overriding the base url

### CODE REVIEW

File: mongodbatlas/config.go [hunk 1]
```diff
@@ -2,24 +2,24 @@ package mongodbatlas
...
 
 // Config struct ...
 type Config struct {
-	PublicKey  string
-	PrivateKey string
-	BaseURL    string
+	PublicKey    string
+	PrivateKey   string
+	BaseURL      string
+	RealmBaseURL string
```
Comment (gssbzn) [inline mongodbatlas/config.go:22]: realm has a different base url, if I override the one for atlas I need to change the one for realm as well and they should not share it

File: mongodbatlas/config.go [hunk 2]
```diff
@@ -28,6 +28,8 @@ type MongoDBClient struct {
 	Config *Config
 }
 
+var ua = "terraform-provider-mongodbatlas/" + version.ProviderVersion
```
Comment (gssbzn) [inline mongodbatlas/config.go:31]: extracting the duplicated UA

File: mongodbatlas/config.go [hunk 3]
```diff
@@ -63,12 +65,12 @@ func (c *Config) NewClient(ctx context.Context) (interface{}, diag.Diagnostics)
 func (c *MongoDBClient) GetRealmClient(ctx context.Context) (*realm.Client, error) {
 	// Realm
 	if c.Config.PublicKey == "" && c.Config.PrivateKey == "" {
-		return nil, fmt.Errorf("please set `public_key` and `private_key` in order to use the realm client")
+		return nil, errors.New("please set `public_key` and `private_key` in order to use the realm client")
```
Comment (gssbzn) [inline mongodbatlas/config.go:68]: no need to format as there's no interpolation

File: mongodbatlas/config.go [hunk 4]
```diff
@@ -63,12 +65,12 @@ func (c *Config) NewClient(ctx context.Context) (interface{}, diag.Diagnostics)
...
 	}
 
-	optsRealm := []realm.ClientOpt{realm.SetUserAgent("terraform-provider-mongodbatlas/" + version.ProviderVersion)}
-	if len(c.Config.BaseURL) > 0 {
-		optsRealm = append(optsRealm, realm.SetBaseURL(c.Config.BaseURL))
+	optsRealm := []realm.ClientOpt{realm.SetUserAgent(ua)}
+	if c.Config.BaseURL != "" && c.Config.RealmBaseURL != "" {
```
Comment (gssbzn) [inline mongodbatlas/config.go:72]: for realm you should override both

File: mongodbatlas/provider.go [hunk 5]
```diff
@@ -22,22 +21,38 @@ func Provider() *schema.Provider {
...
 		Schema: map[string]*schema.Schema{
 			"public_key": {
-				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("MONGODB_ATLAS_PUBLIC_KEY", ""),
+				Type:     schema.TypeString,
+				Required: true,
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{
+					"MONGODB_ATLAS_PUBLIC_KEY",
+					"MCLI_PUBLIC_API_KEY",
```
Comment (gssbzn) [inline mongodbatlas/provider.go:28]: pick first non empty

File: mongodbatlas/provider.go [hunk 6]
```diff
@@ -22,22 +21,38 @@ func Provider() *schema.Provider {
...
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("MONGODB_ATLAS_PRIVATE_KEY", ""),
+				Type:     schema.TypeString,
+				Required: true,
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{
+					"MONGODB_ATLAS_PRIVATE_KEY",
+					"MCLI_PRIVATE_API_KEY",
+				}, ""),
 				Description: "MongoDB Atlas Programmatic Private Key",
+				Sensitive:   true,
```
Comment (gssbzn) [inline mongodbatlas/provider.go:40]: [q] any reason we were not setting this? can't find any docs that says not to do it when is a configuration

File: mongodbatlas/provider.go [hunk 7]
```diff
@@ -129,12 +144,10 @@ func Provider() *schema.Provider {
 
 func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
 	config := Config{
-		PublicKey:  d.Get("public_key").(string),
-		PrivateKey: d.Get("private_key").(string),
-	}
-
-	if baseURL, ok := d.GetOk("base_url"); ok {
```
Comment (gssbzn) [inline mongodbatlas/provider.go:136]: no need to `if` here we already `if` to not use it if empty

File: mongodbatlas/provider.go [hunk 8]
```diff
@@ -302,14 +315,3 @@ func HashCodeString(s string) int {
 	// v == MinInt
 	return 0
 }
-
-// HashCodeStrings hashes a list of strings to a unique hashcode.
-func HashCodeStrings(hashStrings []string) string {
```
Comment (gssbzn) [inline mongodbatlas/provider.go:307]: deadcode

File: website/docs/index.html.markdown [hunk 9]
```diff
@@ -8,11 +8,13 @@ description: |-
 
 # MongoDB Atlas Provider
 
-The MongoDB Atlas provider is used to interact with the resources supported by [MongoDB Atlas](https://www.mongodb.com/cloud/atlas). The provider needs to be configured with the proper credentials before it can be used.
+You can use the MongoDB Atlas provider to interact with the resources supported by [MongoDB Atlas](https://www.mongodb.com/cloud/atlas).
```
Comment (gssbzn) [inline website/docs/index.html.markdown:11]: avoiding passive voice

File: website/docs/index.html.markdown [hunk 10]
```diff
@@ -22,37 +24,25 @@ provider "mongodbatlas" {
...
-  private_key  = "atlas_private_api_key" #required
-}
-```
-
-~> *IMPORTANT* Hard-coding your MongoDB Atlas programmatic API key pair into a Terraform configuration is not recommended.  Consider the risks, especially the inadvertent submission of a configuration file containing secrets to a public repository.
+The MongoDB Atlas provider offers a flexible means of providing credentials for authentication. 
+You can use any the following methods:
 
-### Environment variables
+### Environment Variables
```
Comment (gssbzn) [inline website/docs/index.html.markdown:41]: env first, static later, lazy readers will stop after first example and may skip the warning is not secure

File: mongodbatlas/provider.go (hunk 6 above)
Comment (themantissa) [reply mongodbatlas/provider.go:40]: Good question, I'm not aware of any blocker either but @abner-dou @coderGo93 or @thetonymaster - any issue?

**Review by themantissa — APPROVED** [review]
A nice add as we do have users with both.  Also appreciate the other improvements.
LGTM!

File: mongodbatlas/provider.go (hunk 6 above)
Comment (coderGo93) [reply mongodbatlas/provider.go:40]: Good eye, yes for private key should be a sensitive, I wasn't aware of that part probably it was there that from the beginning

File: mongodbatlas/config.go (hunk 1 above)
Comment (coderGo93) [reply mongodbatlas/config.go:22]: oh makes sense, thank you gustavo

**Review by coderGo93 — APPROVED** [review]
LGTM

**Pull request merged**